tests/heartbeat.test.js

```diff
@@ -83,12 +83,15 @@ describe('heartbeat', () => {
         });
         Object.values(supervisor.workers).forEach((worker) => {
           worker.emit('message', { act: 'poolHallHeartbeat' });
-          worker.emit('up');
+          worker.process.emit('message', { poolHallInternal: true, act: 'ready' });
         });
       });
 
       it('reports stalled workers', (done) => {
         supervisor.start();
+        Object.values(supervisor.workers).forEach((worker) => {
+          worker.state = 'up'; // eslint-disable-line no-param-reassign
```

Comment:
should this be done with sinon, perhaps, so it can be reset after/before each test?

Reply:
supervisor is constructed per test so there is no leakage, state transitions necessarily mutate the state variable